tests/BenchmarkDotNet.Tests/ParameterComparerTests.cs

```diff
@@ -157,6 +157,70 @@ public void IComparableComparisionTest()
             Assert.Equal(4, ((ComplexParameter)sortedData[3].Items[0].Value).Value);
         }
 
+        [Fact]
+        public void ValueTupleWithNonIComparableInnerTypesComparisionTest()
+        {
+            var comparer = ParameterComparer.Instance;
+            var originalData = new[]
+            {
+                new ParameterInstances(new[]
+                {
+                    new ParameterInstance(sharedDefinition, (new ComplexNonIComparableParameter(), 1), null)
```

Comment:
Does this test fail without the try/catch? The exception message indicates that it would succeed since `int` is `IComparable`. I would expect it to only fail if _all_ types are non-`IComparable`.

Reply:
Yes, this fails without the try/catch.

> I would expect it to only fail if _all_ types are non-`IComparable`.

Not exactly. The exception message doesn't come from Tuple/ValueTuple, it's from [System.Collections.Comparer.Compare](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Collections/Comparer.cs,67), and it's talking about one of the two values in the arguments.

In the `ValueTuple`'s [implementation](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/ValueTuple.cs,724) for example, it's just calling the comparer on each item. So `CompareTo` could work if int was the first item, but only if the compare is non 0.